### ISSUE TITLE
Feature request: Provide a way to delegate back to gulp-rewrite-css internal adapt path function

### ISSUE THREAD

**Issue by mattbrooks2010** [joscha/gulp-rewrite-css#69]
I have a scenario where I want to use `gulp-rewrite-css` to rewrite some CSS paths and not others. In the former cases, I would like to be able to leverage the internal / built-in adapt path functionality in `gulp-rewrite-css`.

The code I have (below) currently has to duplicate the built-in logic.

```js
.pipe(rewriteCss({
    destination: "./Bundles",
    adaptPath: function (ctx) {
        const sourceFileRelativePath = path.relative(".", ctx.sourceFile);
        const isSourceFileBundled = mm.any(sourceFileRelativePath, paths.cssBundles);

        if (isSourceFileBundled) {
            // Would be nice to do something like `return ctx.originalAdaptPathFn;` instead.
            return path.join(path.relative(ctx.destinationDir, ctx.sourceDir), ctx.targetFile);
        }

        // Leave original path unchanged from source.
        return undefined;
    }
}))
```

It would be nice to be able to do something like `return ctx.originalAdaptPathFn;` instead, or otherwise return some kind of flag back to `gulp-rewrite-css`.

**Comment by joscha** [comment]
can you open a PR to show how this would change the code?

**Comment by mattbrooks2010** [comment]
That could be difficult for me — I'm not really setup for CoffeeScript. I could post a snippet of the updated compiled JavaScript here, if that would help?

The code snippet in the sample above `path.join(path.relative(ctx.destinationDir, ctx.sourceDir), ctx.targetFile)` is copied from your source, and is something I need to duplicate in my scenario. It would be nice to do something like:

```js
.pipe(rewriteCss({
    destination: "./Bundles",
    adaptPath: function (ctx) {
        // Logic specific to my scenario.
        const sourceFileRelativePath = path.relative(".", ctx.sourceFile);
        const isSourceFileBundled = mm.any(sourceFileRelativePath, paths.cssBundles);

        if (isSourceFileBundled) {
            // Here I want to use functionality already present in your package (not supported).
            return ctx.originalAdaptPathFn(ctx);
        }

        // Here I want to leave original path unchanged from source (already supported).
        return undefined;
    }
}))
```

**Comment by mattbrooks2010** [comment]
Changes like the below would work for me and should be non-breaking. It only requires 3 changes, which with comments.

I'm not fixated on the actual name used for the additional property `originalAdaptPathFn `.

```js
  module.exports = function(opt) {
    var bufferReplace, logRewrite, mungePath, rewriteUrls, streamReplace;
    // Change 1: Always create the internal adaptPath function.
    var adaptPath = function(ctx) {
      return path.join(path.relative(ctx.destinationDir, ctx.sourceDir), ctx.targetFile);
    };
    if (opt == null) {
      opt = {};
    }
    if (opt.debug == null) {
      opt.debug = false;
    }
    if (opt.adaptPath == null) {
      // Change 2: User has not specified adaptPath override.
      opt.adaptPath = adaptPath;
    }
```

```js
    mungePath = function(match, sourceFilePath, file) {
      var destinationDir, sourceDir, targetUrl;
      if ((isRelativeUrl(file)) && !(isRelativeToBase(file))) {
        destinationDir = opt.destination;
        sourceDir = path.dirname(sourceFilePath);
        targetUrl = opt.adaptPath({
          sourceDir: sourceDir,
          sourceFile: sourceFilePath,
          destinationDir: destinationDir,
          targetFile: file,
          // Change 3: Always pass reference to internal adaptPath function.
          originalAdaptPathFn: adaptPath
        });
```

Or can something like this already be achieved?

**Comment by joscha** [comment]
how about exporting the original `adaptPath` function (that is currently on `opts` and overridden when you pass a custom one) separately, so you can use it via `require('gulp-rewrite-css').adaptPath` - would that fix your use-case?

**Comment by mattbrooks2010** [comment]
Yes, that would work. My only requirement is I need access to that function. Thanks!

**Comment by joscha** [comment]
released in `gulp-rewrite-css@1.1.0`

**Comment by mattbrooks2010** [comment]
Great, thanks!
